### ISSUE TITLE
How to send errors back to the client?

### ISSUE THREAD

**Issue by MrMage** [grpc/grpc-swift#123]
(I've found #10 and #88 on this, but no further information, and not sure whether these are referring to this.)

Looking at the generated server code (see [`Plugin/Templates/server.swift`](https://github.com/grpc/grpc-swift/blob/master/Plugin/Templates/server.swift)) surfaces the following handler code:

      do {
        switch handler.method {
        case "/service/operation":
          try Service_OperationSession(handler:handler, provider:provider).run(queue:queue)
        default:
          // handle unknown requests
          try handler.receiveMessage(initialMetadata:Metadata()) {(requestData) in
            try handler.sendResponse(statusCode:.unimplemented,
                                     statusMessage:"unknown method " + handler.method,
                                     trailingMetadata:Metadata())
          }
        }
      } catch (let error) {
        print("Server error: \(error)")
      }

It appears that errors thrown in `try Service_OperationSession(handler:handler, provider:provider).run(queue:queue)` simply never return a result whatsoever to the client, instead of returning the appropriate `StatusCode`. Should this be changed, to at least let the client know that something bad happened instead of just going radio silent? It might also be nice to have the option to e.g. return custom status codes or messages to the client. A simple approach could simply try sending a response with an error status code in the catch block.

**Comment by MrMage** [comment]
Errors will be sent back properly once https://github.com/grpc/grpc-swift/pull/184 is merged.

**Comment by MrMage** [comment]
Done with #184.
